### PR TITLE
Generate a name for AI chat based on the conversation

### PR DESCRIPTION
**Pull request by ravikiranvm**
Fixes OPS-2172.

## Additional Notes
- AI Chat name generates after the 3rd message from the user. 
- Once a chat name is generated, it stays static until the end of conversation. 

- Why I chose 3 user messages and not more? I believe the first couple of messages the user might not send the "actual" questions but just something generic like "Hey, I need your help". Also I thought it is better to not wait for more context, say more than 5 messages from the user, for better UX. 

## Testing Checklist

Check all that apply:

- [x] I tested the feature thoroughly, including edge cases

- [x] I verified all affected areas still work as expected

- [x] Automated tests were added/updated if necessary

- [x] Changes are backwards compatible with any existing data, otherwise a migration script is provided

## Visual Changes (if applicable)

If there are UI/UX changes, include at least one of the following:

- Screenshots
- Loom video
- Preview deployment link

### CODE REVIEW

File: packages/server/api/src/app/ai/chat/ai-chat.service.ts [hunk 1]
```diff
@@ -37,8 +37,46 @@ export type MCPChatContext = {
...
+};
+
+export async function generateChatName(
+  messages: CoreMessage[],
+  projectId: string,
+): Promise<string> {
+  const { languageModel } = await getLLMConfig(projectId);
+  const prompt: CoreMessage[] = [
+    {
+      role: 'assistant',
```
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/server/api/src/app/ai/chat/ai-chat.service.ts]: Fixed

File: packages/server/api/test/unit/ai/ai-mcp-chat.controller.test.ts [hunk 2]
```diff
@@ -470,8 +486,7 @@ describe('AI MCP Chat Controller - Tool Service Interactions', () => {
       });
     });
 
-    it('should not call tools and load only promt', async () => {
-      (getMCPTools as jest.Mock).mockResolvedValue(mockAllTools);
+    it('should not call tools and load only prompt', async () => {
```
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/server/api/test/unit/ai/ai-mcp-chat.controller.test.ts:478]: It was already fixed

File: packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts [hunk 3]
```diff
@@ -174,7 +196,6 @@ export const aiMCPChatController: FastifyPluginAsyncTypebox = async (app) => {
           projectId,
         );
 
-        mcpClients = toolSet.mcpClients;
         mcpClients = toolSet.mcpClients;
```
[the remark this answers is not in the document]
Comment (ravikiranvm) [reply packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts:199]: It was already fixed

**Comment by ravikiranvm** [comment]
@rSnapkoOpenOps I've deployed the changes to UX.

File: packages/server/api/src/app/ai/chat/ai-chat.service.ts [hunk 4]
```diff
@@ -37,8 +37,46 @@ export type MCPChatContext = {
...
+    {
+      role: 'system',
+      content: `
+      You are an AI assistant on the OpenOps platform, where users interact about FinOps, cloud providers (AWS, Azure, GCP), OpenOps features, and workflow automation.
+      Given the following conversation, suggest a short, descriptive name (max five words) that best summarizes the main topic, question, or action discussed in this chat. 
+      The name should be specific (not generic like "Chat" or "Conversation"), and reflect the user's intent (e.g., "AWS Cost Optimization", "Create Budget Workflow", "OpenOps Integration Help").
+      Limit the name to five words or less.
+      Respond with only the name.
+      `,
+    } as const,
```
Comment (rmn-snpk) [inline packages/server/api/src/app/ai/chat/ai-chat.service.ts]: We store prompts in `ai-prompts` folder, and fetch them using `loadPrompt`. Would be cool to extract it

File: packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts [hunk 5]
```diff
@@ -155,6 +160,23 @@ export const aiMCPChatController: FastifyPluginAsyncTypebox = async (app) => {
...
+        projectId,
+      );
+      if (
+        (updatedContext.userMessageCount ?? 0) >= 3 &&
+        !updatedContext.chatName
+      ) {
+        const name = await generateChatName(
+          conversationResult.messages,
+          projectId,
+        );
```
Comment (rmn-snpk) [inline packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts]: can we use `conversationResult` to check the length of the conversation?

File: packages/server/api/src/app/ai/chat/ai-chat.service.ts (hunk 4 above)
Comment (ravikiranvm) [reply packages/server/api/src/app/ai/chat/ai-chat.service.ts]: Done :)

File: packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts (hunk 5 above)
Comment (ravikiranvm) [reply packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts]: Yes, now I've removed the unnecessary incrementUserMessageCount func :)
